src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,39 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
```

Comment:
It is based on that comment and I've asked him to provide these functions. I agree they are not strictly needed, but providing them is good for multiple reasons:
 - other bindings can use the same mapping, helping our users with consistency;
 - even in C, sometimes you want to debug or print something to the user, then having a enum<->string is helpful; Like you expose this enum via a OIC or HTTP/JSON.

These functions should be provided by soletta's C API, we missed them but I count on you guys to provide these as you notice they are missing. If you don't have time you can fill a ticket for others to tackle them.